### DISCUSSION TITLE
gathered metric family process_open_fds has type UNTYPED but should have GAUGE

### DISCUSSION THREAD

**Discussion by YuyueBalabala** [prometheus/pushgateway#602] [Q&A]
I use http request and push metric to pushgateway ,and it return 400.shows
`36 error(s) occurred:\n* 
gathered metric family process_open_fds has type UNTYPED but should have GAUGE
gathered metric family go_memstats_heap_inuse_bytes has type UNTYPED but should have GAUGE
gathered metric family go_memstats_heap_released_bytes has type UNTYPED but should have GAUGE
gathered metric family go_memstats_last_gc_time_seconds has type UNTYPED but should have GAUGE
gathered metric family go_memstats_alloc_bytes has type UNTYPED but should have GAUGE
collected metric named \"go_gc_duration_seconds_sum\" collides with previously collected summary named \"go_gc_duration_seconds\"
gathered metric family go_memstats_mspan_inuse_bytes has type UNTYPED but should have GAUGE\n* gathered metric family go_info has type UNTYPED but should have GAUGE...`

I don't know what's wrong with my metrcis data.

**Comment by beorn7** [comment]
These are all metrics that the Pushgateway also exposes about itself. You can still push those metrics (although that's not recommended) but you have to make sure to push them with the same type. For example the `process_open_fds` metric is exposed by the PGW itself as a GAUGE but now you try to push it as UNTYPED.

You have to fix the program pushing the metrics to use the correct type for the pushed metrics. Or don't push those metrics at all. Remember, the PGW is not meant to turn Prometheus into a push-based monitoring system. You should usually push a handful of selected metrics from a batch job or something rather than pushing all of the metrics you would usually expose for Prometheus to scrape.

If you want to mass-push metrics data, you should better try options like using the Prometheus remote-write protocol or OTLP from the OpenTelemetry ecosystem.
